### PR TITLE
snap: support for parsing and exposing on snap.Info aliases

### PR DESCRIPTION
**Pull request by pedronis**
This adds support for parsing and exposing on snap.Info aliases lists on snap applications.

### CODE REVIEW

File: snap/info_snap_yaml.go [hunk 1]
```diff
@@ -218,12 +223,13 @@ func setSlotsFromSnapYaml(y snapYaml, snap *Info) error {
 	return nil
 }
 
-func setAppsFromSnapYaml(y snapYaml, snap *Info) {
+func setAppsAndAliasesFromSnapYaml(y snapYaml, snap *Info) error {
```
Comment (niemeyer) [inline snap/info_snap_yaml.go]: New function name feels oververbose. Keeping the old one but including the new logic would seem alright.

File: snap/info_snap_yaml.go [hunk 2]
```diff
@@ -243,6 +249,12 @@ func setAppsFromSnapYaml(y snapYaml, snap *Info) {
 			app.Slots = make(map[string]*SlotInfo)
 		}
 		snap.Apps[appName] = app
+		for _, alias := range app.Aliases {
+			if snap.Aliases[alias] != nil {
+				return fmt.Errorf("cannot set %q as alias for both %q and %q", alias, snap.Aliases[alias].Name, appName)
```
Comment (niemeyer) [inline snap/info_snap_yaml.go]: Nice.

File: snap/validate.go [hunk 3]
```diff
@@ -84,6 +86,13 @@ func Validate(info *Info) error {
 		}
 	}
 
+	// validate aliases
+	for alias, app := range info.Aliases {
+		if !validAlias.MatchString(alias) {
+			return fmt.Errorf("cannot have %q as alias name for app %q - use letters, digits, and dash, underscore or dot as separator", alias, app.Name)
```
Comment (niemeyer) [inline snap/validate.go]: It's not actually a separator in this case. It's the alias name itself.

**Comment by mvo5** [comment]
:+1:

**Pull request merged**